modules/media/jsx/uploadForm.js

```diff
@@ -215,12 +216,21 @@ var MediaUploadForm = React.createClass({
     var requiredFileName = this.getValidFileName(
       myFormData.pscid, myFormData.visitLabel, instrument
     );
-
     if (!this.isValidFileName(requiredFileName, fileName)) {
       alert("File name should begin with: " + requiredFileName);
       return;
     }
 
+    // Check for duplicate file names
+    var isDuplicate = mediaFiles.indexOf(myFormData.file.name);
+    if (isDuplicate >= 0) {
+      var confirmed = confirm(
+        "A file with this name already exists!\n" +
```

Comment:
I was using the unique index on file name in addition to the changes in this pull request.